src/Symfony/Component/VarExporter/Internal/Configurator.php

```diff
@@ -99,15 +97,33 @@ public static function getConfigurator($class)
                 };
         }
 
+        $privateProperties = 0;
         $propertyReflectors = array();
         foreach ($classReflector->getProperties(\ReflectionProperty::IS_PROTECTED | \ReflectionProperty::IS_PRIVATE) as $propertyReflector) {
             if (!$propertyReflector->isStatic()) {
                 $propertyReflector->setAccessible(true);
                 $propertyReflectors[$propertyReflector->name] = $propertyReflector;
+                $privateProperties += $propertyReflector->isPrivate();
             }
         }
 
-        return self::$configurators[$class] = static function ($properties, $objects) use ($propertyReflectors) {
+        if (!$propertyReflectors || !($privateProperties || $classReflector->isFinal())) {
+            self::$configurators[$class] = static function ($properties, $objects) {
+                foreach ($properties as $name => $values) {
+                    foreach ($values as $i => $v) {
+                        $objects[$i]->$name = $v;
+                    }
+                }
+            };
+
+            if ($propertyReflectors) {
+                self::$configurators[$class] = \Closure::bind(self::$configurators[$class], null, eval('return new class() extends '.$class.'{};'));
```

Comment:
It is common for hosts to disable `eval` for security reasons. Is it possible to keep the less optimized code for such case (checked using `function_exists('eval')`) ? Or maybe to avoid using `eval` entirely ?

Reply:
eval is not a function, it's a language construct
how can you disable it?

Reply:
hmm, indeed, disabling it is not possible through php.ini. It was the suhosin extension disabling it. Not sure how this gets detected then

Reply:
I actually figured out a way to not need eval() and still make the dump more compact are fast.